modules/table/src/main/java/org/apache/ignite/internal/table/InternalTable.java

```diff
@@ -296,6 +297,31 @@ Publisher<BinaryRow> scan(
             @Nullable BitSet columnsToInclude
     );
 
+    /**
+     * Scans given partition index, providing {@link Publisher} that reactively notifies about partition rows.
+     *
+     * @param partId The partition.
+     * @param txId Transaction id.
+     * @param leaderNode Raft group leader node that must handle given get request.
+     * @param leaderTerm Raft group leader term.
+     * @param lowerBound Lower search bound.
+     * @param upperBound Upper search bound.
+     * @param flags Control flags. See {@link org.apache.ignite.internal.storage.index.SortedIndexStorage} constants.
+     * @param columnsToInclude Row projection.
+     * @return {@link Publisher} that reactively notifies about partition rows.
+     */
+    Publisher<BinaryRow> scan(
+            int partId,
+            UUID txId,
+            ClusterNode leaderNode,
+            long leaderTerm,
```

Comment:
Leader and term need to be wrapped to PrimaryReplica class.

Reply:
Done, thanks